Framework/NexusGeometry/inc/MantidNexusGeometry/InstrumentGeometryAbstraction.h

```diff
@@ -0,0 +1,40 @@
+#ifndef INSTRUMENT_ABSTRACT_BUILDER_H
+#define INSTRUMENT_ABSTRACT_BUILDER_H
+
+//----------------------
+// Includes
+//----------------------
+
+#include "MantidGeometry/Instrument_fwd.h"
+#include "MantidGeometry/Instrument/ObjCompAssembly.h"
+#include "MantidNexusGeometry/InstrumentAbstractBuilder.h"
+#include "MantidNexusGeometry/ShapeGeometryAbstraction.h"
+
+#include "Eigen/Core"
+#include <string>
+
+namespace Mantid{
+namespace NexusGeometry{
+
+class DLLExport InstrumentGeometryAbstraction : public NexusGeometry::InstrumentAbstractBuilder<InstrumentGeometryAbstraction>{
```

Comment:
Name of this type needs to change. This is no longer an "Abstraction". `InstrumentBuilder` would be fine. Maybe we'll call the `Instrument2.0` one `BeamlineBuilder`